honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,330 @@
+
+    return faces
+
+
+def extract_mesh_faces_colors(mesh, color_by_face=False):
+    """Extract face indices and colors from a Rhino mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A tuple of mesh faces and mesh colors
+    """
+
+    colors = None
+    lb_faces = []
+    for i in range(len(mesh.Faces)):
+        face = mesh.Faces[i]
+        if len(face) == 4:
+            lb_faces.append((face[0], face[1], face[2], face[3]))
+        else:
+            lb_faces.append((face[0], face[1], face[2]))
+    if len(mesh.VertexColors) != 0:
+        colors = []
+        if color_by_face is True:
+            for j in range(len(mesh.Faces)):
+                face = mesh.Faces[j]
+                col = mesh.VertexColors[face[0]]
+                colors.append(lbc.Color(col.R, col.G, col.B))
+        else:
+            for k in range(len(mesh.VertexColors)):
+                col = mesh.VertexColors[k]
+                colors.append(lbc.Color(col.R, col.G, col.B))
+    return lb_faces, colors
+
+
+def mesh_to_mesh3d(mesh, color_by_face=True):
+    """Get a Ladybug Mesh3D object from a Rhino3dm Mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A Ladybug Mesh3D object
+    """
+    lb_verts = tuple(to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices)))
+    lb_faces, colors = extract_mesh_faces_colors(mesh, color_by_face)
+    return Mesh3D(lb_verts, lb_faces, colors)
+
+
+def brep_to_face3d(brep):
+    """Get a list of Ladybug Face3D objects from a rhino3dm Brep.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+    return faces
+
+
+def check_planarity(brep):
+    """Check the planarity of a Brep.
+
+    Args:
+        brep: Geometry of a Rhino3dm Brep
+
+    Returns:
+        Bool. True if planar otherwise False
+    """
+    is_planar = [brep.Faces[i].UnderlyingSurface().IsPlanar()
+        for i in range(len(brep.Faces))]
+    return all(is_planar)
+
+
+def brep2d_to_face3d(brep, tolerance):
+    """Get a list of Ladybug Face3D objects from a planar rhino3dm Brep.
+
+    This function is used in translating rhino3dm planar Brep to Ladybug Face3D objects.
+
+    Args:
+        brep: A rhino3dm Brep.
+        tolerance: A rhino3dm tolerance object. Tolerance set in the rhino file.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    # * Check 01 - If any of the edge is an arc
+    lines = []
+
+    # Getting all vertices of the face
+    mesh = brep.Faces[0].GetMesh(rhino3dm.MeshType.Any)
+    mesh_pts = [to_point3d(mesh.Vertices[i])
+                for i in range(len(mesh.Vertices))]
```

Comment:
It looks like you don't end up using these points at all so I would just get rid of this variable.  You can replace it with a check of `len(mesh.Vertices)` later in the script.

Reply:
Complied.